Docs: clarify GCP service accounts. (#2761)
skypilot-org/skypilot: docs/source/cloud-setup/cloud-auth.rst

GCP Service Account
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

`GCP Service Account
<https://cloud.google.com/iam/docs/service-account-overview>`__ is supported.

.. tip::
    A long-lived service account on your local machine can avoid the periodic
    ``google.auth.exceptions.RefreshError: Reauthentication is needed. Please
    run `gcloud auth application-default login` to reauthenticate.`` error.

Set up a service account as follows:

1. Follow the :ref:`instructions <gcp-service-account-creation>` to create a service account with the appropriate roles/permissions.
2. In the "Service Accounts" tab in the `IAM & Admin console
   <https://console.cloud.google.com/iam-admin/iam>`__, click on the service
   account to go to its detailed page. Click on the **KEYS** tab, then click on

>    **ADD KEY** to add a JSON key.  The key will be downloaded automatically.

Reviewer: While creating the key, I was trying to look for an option to make it "long-lived" (as suggested by the tip above). Turns out service account keys that you create and download from IAM don't have an expiry time and stay valid until you delete them. Might be worth mentioning

Author: Added a clarification.

3. Set the environment variable ``GOOGLE_APPLICATION_CREDENTIALS`` to the path of the key file, and configure the gcloud CLI tool:

   .. code-block:: console

       $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/key.json
       $ gcloud auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
       $ gcloud config set project your-project-id

>    You may want to add the export statement in your profile (e.g. ``~/.bashrc``, ``~/.zshrc``) so that it is set automatically in any terminal window.

Reviewer: ```suggestion
   You may want to add the export statement in your profile (e.g. ``~/.bashrc``, ``~/.zshrc``) so that it is set automatically in all new terminal sessions.
```
